package.json

```diff
@@ -3,12 +3,13 @@
   "version": "0.16.3",
   "description": "The Phoenix LiveView JavaScript client.",
   "license": "MIT",
-  "main": "./priv/static/phoenix_live_view.esm.js",
   "module": "./priv/static/phoenix_live_view.esm.js",
+  "main": "./priv/static/phoenix_live_view.cjs.js",
   "unpkg": "./priv/static/phoenix_live_view.min.js",
   "jsdelivr": "./priv/static/phoenix_live_view.min.js",
   "exports": {
-    "import": "./priv/static/phoenix_live_view.esm.js"
```

Comment:
I think this is right but I wonder if we should be providing a "module" here as well (see https://webpack.js.org/guides/package-exports/#providing-commonjs-and-esm-version-stateless)

Reply:
I think so?
I did some testing and at least for my purposes, esbuild likes to default to using the "iife" (immediately invoked function expression) format, and wraps the `cjs` file so it can do an "import" at runtime. I was only able to force esbuild to use the `esm.js` by setting it as the `"main"`, which is an odd `esbuild` thing? (https://github.com/evanw/esbuild/issues/840)

Ultimately though, having the `"module"` at the top level looks like it is potentially helpful in certain scenarios?
The `"main": "...cjs.js"` works for Yarn2 + esbuild when following the docs, and assuming there is a way to tell esbuild to prefer the `esm.js` file over the `cjs`, it worked just fine then too.


From the general gist, it seems like leaving the "module" and just adding the "type" is the typical thing?
```
{
  "type": "module",
  "module": "./priv/static/phoenix.esm.js",
  "main": "./priv/static/phoenix.cjs.js",
  "unpkg": "./priv/static/phoenix.min.js",
  "jsdelivr": "./priv/static/phoenix.min.js",
  "exports": {
    "import": "./priv/static/phoenix.esm.js",
    "require": "./priv/static/phoenix.cjs.js"
  }
}
```

I thoroughly dislike JS dependency resolution now 😄

Reply:
I am leaving this unresolved for posterity :)